server.go

```diff
@@ -467,7 +467,16 @@ func (s *Server) inheritClientSession(pk packets.Packet, cl *Client) bool {
 			return false // [MQTT-3.2.2-3]
 		}
 
-		cl.State.Inflight = existing.State.Inflight // [MQTT-3.1.2-5]
+		infLen := existing.State.Inflight.Len()
+		if infLen > 0 {
+			cl.State.Inflight = existing.State.Inflight // [MQTT-3.1.2-5]
+			if cl.State.Inflight.maximumReceiveQuota == 0 && cl.ops.capabilities.ReceiveMaximum != 0 {
+				cl.State.Inflight.ResetReceiveQuota(int32(cl.ops.capabilities.ReceiveMaximum)) // server receive max per client
+				cl.State.Inflight.ResetSendQuota(int32(cl.Properties.Props.ReceiveMaximum))    // client receive max
+				cl.State.Inflight.receiveQuota -= int32(infLen)                                // available receive quota is equal to the maximum quota minus the number of inflight messages
```

Comment:
Ah, sorry for the delay @wind-c - it's been a very busy week. I understand what you are trying to do now you have explained it, and I believe it is correct. The code looks good.

However, my one concern is still this line:

```go
cl.State.Inflight.receiveQuota -= int32(infLen)  // available receive quota is equal to the maximum quota minus the number of inflight messages`
```

Even though there are inflights still pending from the existing, the receive quota should be reset across connections (as specified in the spec). Therefore, we should remove this final line.

What do you think?

Reply:
ok, so remove this line.

Reply:
@wind-c Yes, if this line is removed, I believe we can merge this PR. Apologies again for the delay in replying 🙏🏻